src/plugins/auto/plugin.cpp

```diff
@@ -329,18 +329,22 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
     // updateFromMap will check config valid
     loadConfig.UpdateFromMap(config, GetName(), true);
     auto fullConfig = loadConfig._keyConfigMap;
+    bool workModeAuto = GetName() == "AUTO";
     // Remove the performance hint if no setting to this property from user.
     if (!loadConfig._isSetPerHint) {
         fullConfig.erase(PluginConfigParams::KEY_PERFORMANCE_HINT);
-        // set performance hint to 'THROUGHPUT' model for AutoExecutable Network.
         loadConfig._perfHintsConfig.SetConfig(PluginConfigParams::KEY_PERFORMANCE_HINT, PluginConfigParams::THROUGHPUT);
+        if (workModeAuto) {
```

Comment:
please also check if user set performance hint during compile_model()

Reply:
The above code line 330 has been checked if performance hint is set.
`loadConfig.UpdateFromMap(config, GetName(), true);`